examples/train_from_custom_env.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 OmniSafe Team. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Example and template for environment customization."""
+
+from __future__ import annotations
+
+import random
+from typing import Any, ClassVar
+
+import torch
+from gymnasium import spaces
+
+import omnisafe
+from omnisafe.envs.core import CMDP, env_register
+
+
+# First, define the environment class.
+# The most important thing is to add the `env_register` decorator.
+@env_register
+class CustomExampleEnv(CMDP):
+    _support_envs: ClassVar[list[str]] = ['Custom-v0']
```

Comment:
Maybe we can add more detailed explanations about what the key components mean (e.g.`_support_envs` ) when customizing our envs in this file.

Reply:
Thank you for your reminder; I have added a more detailed explanation.